### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by lyuboasenov**
This change: 
* updates newtonsof.json to 13.0.1 (latest) and RestSharp to 106.15.0 (the following versions have breaking change). 
* adds working build procedure for the c# bindings for the srs
* enable building the srs appliance on vcenter server - not documented though

### CODE REVIEW

File: src/Admin/VMware.ScriptRuntimeService.Setup/Program.cs [hunk 1]
```diff
@@ -19,6 +19,29 @@ internal class Program {
 
             var logger = loggerFactory.CreateLogger(typeof(Program));
 
+            // Override Newtonsoft.Json MaxDepth de/serialization because
+            // as of version 13.0.1 the default MaxDepth is 64
+            // this keeps pre 13.0.1 behaviour
+            Func<Newtonsoft.Json.JsonSerializerSettings> defaultSettingsFunc =
+               Newtonsoft.Json.JsonConvert.DefaultSettings;
+            if (null == Newtonsoft.Json.JsonConvert.DefaultSettings) {
```
Comment (atanasa) [inline src/Admin/VMware.ScriptRuntimeService.Setup/Program.cs:27]: Why the need for branching? And why is the NullValueHandling only done in the first branch?

File: src/Endpoint/VMware.ScriptRuntimeService.RunspaceClient/RunspaceClient.cs [hunk 2]
```diff
@@ -17,7 +17,32 @@ namespace VMware.ScriptRuntimeService.RunspaceClient {
...
+            defaultSettingsFunc =
+               () => {
+                  var settings = defaultSettingsFunc();
+                  settings.MaxDepth = int.MaxValue;
+
+                  return settings;
+               };
+         }
+
+         Newtonsoft.Json.JsonConvert.DefaultSettings = defaultSettingsFunc;
```
Comment (atanasa) [inline src/Endpoint/VMware.ScriptRuntimeService.RunspaceClient/RunspaceClient.cs:44]: If code in other classes depend on this global setting, they might not be affected if they don't trigger this static ctor.
Consider placing this in some initialization logic that is explicit or even better - don't use the global settings but pass them where you use the serializers.

File: test/openapi-clients/dotnet/SrsClient/clientsdk/src/IO.Swagger/Client/ApiClient.cs [hunk 3]
```diff
@@ -29,7 +29,8 @@ public partial class ApiClient
     {
         private JsonSerializerSettings serializerSettings = new JsonSerializerSettings
         {
-            ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor
+            ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor,
+            MaxDepth = int.MaxValue
```
Comment (atanasa) [inline test/openapi-clients/dotnet/SrsClient/clientsdk/src/IO.Swagger/Client/ApiClient.cs:33]: Is this code generated? Maybe it should be regenerated instead with a higher version of the generator?

File: test/openapi-clients/dotnet/SrsClient/clientsdk/build.sh [hunk 4]
```diff
@@ -3,66 +3,5 @@
 # Generated by: https://github.com/swagger-api/swagger-codegen.git
```
Comment (atanasa) [inline test/openapi-clients/dotnet/SrsClient/clientsdk/build.sh:3]: If these are not going to be generated anymore, remove the comments section

File: appliance/photon-version.json [hunk 5]
```diff
@@ -9,5 +9,7 @@
   "numvcpus": "2",
   "ramsize": "8192",
   "guest_username": "root",
-  "guest_password": "Srsa!23"
+  "guest_password": "Srsa!23",
+  "guest_network": "VMNetwork",
```
[the remark this answers is not in the document]
Comment (lyuboasenov) [reply appliance/photon-version.json:13]: Yes, these are just values that are injected into the vmware-iso and vsphere-iso - both builders of the hashicorp packer.

File: src/Admin/VMware.ScriptRuntimeService.Setup/Program.cs (hunk 1 above)
Comment (lyuboasenov) [reply src/Admin/VMware.ScriptRuntimeService.Setup/Program.cs:27]: I've removed the null values handling override.

File: src/Endpoint/VMware.ScriptRuntimeService.RunspaceClient/RunspaceClient.cs (hunk 2 above)
Comment (lyuboasenov) [reply src/Endpoint/VMware.ScriptRuntimeService.RunspaceClient/RunspaceClient.cs:44]: The whole library revolves around RunspaceClient and has no other use-cases.

File: test/openapi-clients/dotnet/SrsClient/clientsdk/src/IO.Swagger/Client/ApiClient.cs (hunk 3 above)
Comment (lyuboasenov) [reply test/openapi-clients/dotnet/SrsClient/clientsdk/src/IO.Swagger/Client/ApiClient.cs:33]: There is no new version of the generator, which to bump the newtonsoft json dependency to 13.0.1 and handle the serialization depth issue.

File: BUILD_AND_RUN.md [hunk 6]
```diff
@@ -21,6 +21,16 @@ If you login on the deployed result VM `kubectl` is availble to browse the kuber
 
 ### How to build and run SRS
 
+#### Build SRS
+
+The Photon OS OVF virtual machine is build using HashiCorp Packer. There are two ways of building the virtual machine. You can use `packer` with either the [`vmware-iso`](https://www.packer.io/plugins/builders/vmware/iso) builder or the [`vsphere-iso`](https://www.packer.io/plugins/builders/vsphere/vsphere-iso).
+
+[`vmware-iso`](https://www.packer.io/plugins/builders/vmware/iso) requires ESXi host with running SSH service and `GuestIPHack` enabled.
+
+[`vsphere-iso`](https://www.packer.io/plugins/builders/vsphere/vsphere-iso) requires vSphere Server.
```
Comment (atanasa) [inline BUILD_AND_RUN.md:30]:
```suggestion
[`vsphere-iso`](https://www.packer.io/plugins/builders/vsphere/vsphere-iso) requires vCenter Server.
```

**Review by atanasa — APPROVED** [review]

**Pull request merged**